addons/website_slides/controllers/main.py

```diff
@@ -648,10 +718,115 @@ def _slide_channel_prepare_review_values(self, channel):
 
         return values
 
+    @http.route('/slides/<int:channel_id>/invite', type='http', auth='public', website=True, sitemap=False)
+    def slide_channel_invite(self, channel_id, partner_id=False, invite_hash=False):
+        """ This route is included in the invitation link in email to join / check out the course. It is
+        the main entry point on the attendee's side when sharing or inviting them. As rule of thumb, this will
+        redirect to the course if the rights are given, and to the main /slides page with appropriate error
+        message otherwise. (See _get_invite_error_msg method)
+
+        It acts as a redirector:
+            - A generic invitation link will not have partner_id and invite_hash set. Acl's determine the redirection.
+            - Returns error if parameters are not valid, invitation is more than 3 months old, or if the partner
+            is not in the attendees.
+            - If a user is logged, verify the link is for this user. Redirects according to Acl's.
+            - If no user is logged:
+                - Redirects to login / signup if the partner is enrolled.
+                - Redirects to the course with auth_login / auth_signup depending on whether the partner has a user.
+                They will be able to browse a course preview before logging in / signing up.
+
+        :param channel_id: The id of the course the user is invited to. Do not use <model> in the route instead,
+            otherwise an error 403 could be returned if the (public) user has no access to the record.
+        :param invite_hash: The invitation hash that allows a direct access to channel_id, even if not connected.
+        """
+
+        # --- Parameters
+        partner_id = int(partner_id) or False
+        channel = request.env['slide.channel'].browse(int(channel_id)).exists()
```

Comment:
Yes, actually I removed the use of this route for generic link. I use the direct link to channel with id instead of model, as that route will also redirect to main page if access is denied. It will be computed once, at the start of the route, if public user. :)

Reply:
With this route being used only for direct invitations, we still redirect properly but also redirect to immediate login / signup if they are enrolled.

There is a small duplication in the sense that if 'invited', logged out, and no error, then the check will be made again on the slides/channel_id route.

What do you think?

I must also rewrite the docstring a bit.